packages/flutter_tools/lib/src/commands/build_aot.dart

```diff
@@ -146,5 +198,44 @@ String buildAotSnapshot(
   printStatus('Building snapshot...');
   runCheckedSync(genSnapshotCmd);
 
+  // On iOS, we use Xcode to compile the snapshot into a static library that the
+  // end-developer can link into their app.
+  if (platform == TargetPlatform.ios) {
+    printStatus('Building app.a...');
+
+    // These names are known to from the engine.
+    const String kDartVmIsolateSnapshotBuffer = 'kDartVmIsolateSnapshotBuffer';
+    const String kDartIsolateSnapshotBuffer = 'kDartIsolateSnapshotBuffer';
+
+    runCheckedSync(<String>['mv', vmIsolateSnapshot, path.join(outputDir.path, kDartVmIsolateSnapshotBuffer)]);
+    runCheckedSync(<String>['mv', isolateSnapshot, path.join(outputDir.path, kDartIsolateSnapshotBuffer)]);
+
+    String kDartVmIsolateSnapshotBufferC = path.join(outputDir.path, '$kDartVmIsolateSnapshotBuffer.c');
+    String kDartIsolateSnapshotBufferC = path.join(outputDir.path, '$kDartIsolateSnapshotBuffer.c');
+
+    runCheckedSync(<String>[
+      'xxd', '--include', kDartVmIsolateSnapshotBuffer, kDartVmIsolateSnapshotBufferC
```

Comment:
Yeah, the xxd thing is slightly unfortunate because the C++ symbol it generates is based on the file name.